### PR TITLE
Fix RegExp literal parsing

### PR DESCRIPTION
**Pull request by dbatyai**

### CODE REVIEW

File: jerry-core/parser/js/lexer.cpp [hunk 1]
```diff
@@ -1294,16 +1290,16 @@ lexer_parse_token (void)
...
+      && !(prev_non_lf_token.type == TOK_NAME
+           || prev_non_lf_token.type == TOK_NULL
+           || prev_non_lf_token.type == TOK_BOOL
+           || prev_non_lf_token.type == TOK_CLOSE_BRACE
+           || prev_non_lf_token.type == TOK_CLOSE_SQUARE
+           || prev_non_lf_token.type == TOK_CLOSE_PAREN
+           || prev_non_lf_token.type == TOK_SMALL_INT
+           || prev_non_lf_token.type == TOK_NUMBER
+           || prev_non_lf_token.type == TOK_STRING
+           || prev_non_lf_token.type == TOK_REGEXP))
```
Comment (sand1k) [inline jerry-core/parser/js/lexer.cpp]: `TOK_CLOSE_PAREN` and `TOK_CLOSE_BRACE` could precede regular expression, for example:
```
if (true) /abc/.exec("abc");
if (true) {} /abc/.exec("abc");
```
This case is harder to support, possibly, we could implement it in another patch.

File: jerry-core/parser/js/lexer.cpp [hunk 2]
```diff
@@ -1482,6 +1478,7 @@ lexer_next_token (void)
   if (!is_empty (saved_token))
   {
     sent_token = saved_token;
+    prev_non_lf_token = sent_token;
```
Comment (sand1k) [inline jerry-core/parser/js/lexer.cpp]: `prev_non_lf_token` should be tracked more carefully. It seems, that it should be changed in `lexer_seek` and `lexer_save_token` functions.
For example, the following code:
```
{for (;false;/abc/.exec("abc")) (5)}
```
fails with syntax error due to `prev_non_lf_token` equals to `TOK_CLOSE_BRACE` after call to `lexer_seek` (parser.cpp:1920).

**Comment by dbatyai** [comment]
@sand1k, I've updated the patch, it should work correctly now. Please check.

File: jerry-core/parser/js/lexer.cpp (hunk 1 above)
Comment (dbatyai) [inline jerry-core/parser/js/lexer.cpp]: I agree.

Comment (sand1k) [inline jerry-core/parser/js/lexer.cpp]: Another case:
```
var a
/abc/.exec("abc");
```
Could you, please, create an issue for the mentioned tests?

**Comment by sand1k** [comment]
LGTM.

**Comment by zherczeg** [comment]
LGTM

File: jerry-core/parser/js/lexer.cpp (hunk 1 above)
Comment (dbatyai) [inline jerry-core/parser/js/lexer.cpp]: Issue can be found here: #563

**Pull request merged**